### ISSUE TITLE
Installer for 4.11 fails on MacOS when checking VCenter certificates

### ISSUE THREAD

**Issue by carlosedp** [openshift/installer#6182]
# Version

```console
❯ ./openshift-install version
./openshift-install 4.11.0-rc.6
built from commit 37684309bcb598757c99d3ea9fbc0758343d64a5
release image quay.io/openshift-release-dev/ocp-release@sha256:aaef8e8e0513aa3a6e53a5f7a3761d3e378854e0696612f0ce380c37b0d0be33
release architecture amd64
```

# Platform: VMware vSphere 6.7U3 hosts with vSphere 7.0.3 VCenter

Please specify:
* IPI (automated install with `openshift-install`. If you don't know, then it's IPI)

# What happened?

When trying to install latest RC for OpenShift 4.11 using a MacOS computer as the install machine, I got the error:

```
❯ ./openshift-install create cluster
INFO Consuming Install Config from target directory
INFO Obtaining RHCOS image file from 'https://rhcos.mirror.openshift.com/art/storage/releases/rhcos-4.11/411.86.202207150124-0/x86_64/rhcos-411.86.202207150124-0-vmware.x86_64.ova?sha256='
INFO The file was found in cache: /Users/cdepaula/Library/Caches/openshift-installer/image_cache/rhcos-411.86.202207150124-0-vmware.x86_64.ova. Reusing...
WARNING Discarding the Kubeconfig Admin Client that was provided in the target directory because its dependencies are dirty and it needs to be regenerated
INFO Creating infrastructure resources...
ERROR
ERROR Error: failed to upload: Post "https://192.168.1.14/nfc/52e2e227-df51-d023-2709-8867cbb30565/disk-0.vmdk": x509: “192.168.1.14” certificate is not standards compliant
ERROR
ERROR   with vsphereprivate_import_ova.import,
ERROR   on main.tf line 43, in resource "vsphereprivate_import_ova" "import":
ERROR   43: resource "vsphereprivate_import_ova" "import" {
ERROR
ERROR failed to fetch Cluster: failed to generate asset "Cluster": failure applying terraform for "pre-bootstrap" stage: failed to create cluster: failed to apply Terraform: exit status 1
ERROR
ERROR Error: failed to upload: Post "https://192.168.1.14/nfc/52e2e227-df51-d023-2709-8867cbb30565/disk-0.vmdk": x509: “192.168.1.14” certificate is not standards compliant
ERROR
ERROR   with vsphereprivate_import_ova.import,
ERROR   on main.tf line 43, in resource "vsphereprivate_import_ova" "import":
ERROR   43: resource "vsphereprivate_import_ova" "import" {
ERROR
ERROR
```

This doesn't happen with the 4.10.25 installer.

# What you expected to happen?

Install to progress correctly.

# How to reproduce it (as minimally and precisely as possible)?

Install a standard IPI cluster on vSphere with 4.11 installer from https://mirror.openshift.com/pub/openshift-v4/clients/ocp/

**Comment by jcpowermac** [comment]
@carlosedp can you provide the output: `openssl s_client -showcerts -connect 192.168.1.14:443`

You could also try importing a OVA using `govc`. It uses the same library as the terraform provider.

https://github.com/vmware/govmomi/releases

hmm, maybe this:
https://github.com/golang/go/issues/51991

**Comment by carlosedp** [comment]
Just tried with rc7, same error. Here's the openssl output:

```
❯ openssl s_client -showcerts -connect 192.168.1.14:443
CONNECTED(00000003)
depth=0 C = US, ST = California, L = Palo Alto, O = VMware, OU = VMware Engineering, CN = 192.168.1.14, emailAddress = vmca@vmware.com
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 C = US, ST = California, L = Palo Alto, O = VMware, OU = VMware Engineering, CN = 192.168.1.14, emailAddress = vmca@vmware.com
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/C=US/ST=California/L=Palo Alto/O=VMware/OU=VMware Engineering/CN=192.168.1.14/emailAddress=vmca@vmware.com
   i:/CN=CA/DC=internal/DC=carlosedp.com/C=US/ST=California/O=vcenter.internal.carlosedp.com/OU=VMware Engineering
-----BEGIN CERTIFICATE-----
MIIEVDCCAzygAwIBAgIJAPQdjPw7CoZgMA0GCSqGSIb3DQEBCwUAMIGuMQswCQYD
VQQDDAJDQTEYMBYGCgmSJomT8ixkARkWCGludGVybmFsMR0wGwYKCZImiZPyLGQB
GRYNY2FybG9zZWRwLmNvbTELMAkGA1UEBhMCVVMxEzARBgNVBAgMCkNhbGlmb3Ju
aWExJzAlBgNVBAoMHnZjZW50ZXIuaW50ZXJuYWwuY2FybG9zZWRwLmNvbTEbMBkG
A1UECwwSVk13YXJlIEVuZ2luZWVyaW5nMB4XDTIyMDUwNDE5NTIzNloXDTI3MDUw
NDE5NTIzNlowgZsxCzAJBgNVBAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlhMRIw
EAYDVQQHDAlQYWxvIEFsdG8xDzANBgNVBAoMBlZNd2FyZTEbMBkGA1UECwwSVk13
YXJlIEVuZ2luZWVyaW5nMRUwEwYDVQQDDAwxOTIuMTY4LjEuMTQxHjAcBgkqhkiG
9w0BCQEWD3ZtY2FAdm13YXJlLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
AQoCggEBAKy3oz92bMROyE/cRV8GuqzkH0CFpr1GUQ/aJG1SE7rVrhRwBKZ+d6uM
s9Y/hmhlBEPASAzHptDnuywj0JvkXKxl+7HeBP7Z1KtLjWsTAiQlkq6OAss+8SCm
XYn14TEf0erbL5FA0JtkJCpesBpNnyFTa1vFbHhqPGWMuYsMSulXeDk+7E+1IBp+
vPGFZ+RsrC7k0djdSTQFxJ073/LGb8msfBJ6Dw7KrxJVjutMOIkWFFW1cehdwn/w
lg/rWY1PDsE7YgovOZJIaqbGwvKU0bk1uYmaR2eNtz0+y5byiKYa4RLoalcvFfsJ
hk+/OvkMLxlCUBjIavCSLBmaRNsFDosCAwEAAaOBhTCBgjAPBgNVHREECDAGhwTA
qAEOMB8GA1UdIwQYMBaAFMTDA/JTJ1F1Nil9at9gR2TtBrBdME4GCCsGAQUFBwEB
BEIwQDA+BggrBgEFBQcwAoYyaHR0cHM6Ly92Y2VudGVyLmludGVybmFsLmNhcmxv
c2VkcC5jb20vYWZkL3ZlY3MvY2EwDQYJKoZIhvcNAQELBQADggEBAMPHw399SThU
AySJ8U6yNftVI3ExYWXoLrPQbaQwl/K0vMnDdvaXMAiUCiU56Lq1dl0o4/C0aDhZ
+WlV3h1mXSGycQzKFSeeHCuDyi2OG6PPfiSOKO4Rpl3hxjeaSLx/AVSgyZwvIP/L
oXX0AHoM4wt0NlzJx2E+na1Nu5BNS+yfQ9vZUxghciK5Xzc/mZYeQWhLjxs7qgxm
H//ultKVJaPniiO5rttXa6CJRMjsOoFFEed5pJZr9/on0H9srXwS8fX4x7oVHX/Q
B/HBWD3gi0W2kbopw5zzo383hEvL7SOAMwfaB6NwARtZKKSDZDe0GRlDteahyUl1
QEhEB3w0psw=
-----END CERTIFICATE-----
---
Server certificate
subject=/C=US/ST=California/L=Palo Alto/O=VMware/OU=VMware Engineering/CN=192.168.1.14/emailAddress=vmca@vmware.com
issuer=/CN=CA/DC=internal/DC=carlosedp.com/C=US/ST=California/O=vcenter.internal.carlosedp.com/OU=VMware Engineering
---
No client certificate CA names sent
Server Temp Key: ECDH, P-256, 256 bits
---
SSL handshake has read 1587 bytes and written 322 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES256-GCM-SHA384
    Session-ID:
    Session-ID-ctx:
    Master-Key: 5DBF8235EA76BBE2B4A1F94939FDEFB1E936561727822492ECC685415F4AE13AED50F0BFB3410A626830B00DEF15F327
    Start Time: 1659620098
    Timeout   : 7200 (sec)
    Verify return code: 21 (unable to verify the first certificate)
---
```

As a clarification, my install-config sets the vcenter with the hostname, it has the IP 192.168.1.10. The IP 192.168.1.14 is my ESXi host.

Same openssl command to check my VCenter certificate:

```
❯ openssl s_client -showcerts -connect 192.168.1.10:443
CONNECTED(00000003)
depth=0 CN = vcenter.internal.carlosedp.com, C = US
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN = vcenter.internal.carlosedp.com, C = US
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/CN=vcenter.internal.carlosedp.com/C=US
   i:/CN=CA/DC=internal/DC=carlosedp.com/C=US/ST=California/O=vcenter.internal.carlosedp.com/OU=VMware Engineering
-----BEGIN CERTIFICATE-----
MIIENDCCAxygAwIBAgIJAMUCF78MMUGNMA0GCSqGSIb3DQEBCwUAMIGuMQswCQYD
VQQDDAJDQTEYMBYGCgmSJomT8ixkARkWCGludGVybmFsMR0wGwYKCZImiZPyLGQB
GRYNY2FybG9zZWRwLmNvbTELMAkGA1UEBhMCVVMxEzARBgNVBAgMCkNhbGlmb3Ju
aWExJzAlBgNVBAoMHnZjZW50ZXIuaW50ZXJuYWwuY2FybG9zZWRwLmNvbTEbMBkG
A1UECwwSVk13YXJlIEVuZ2luZWVyaW5nMB4XDTIwMDgyNDEzMjYyOVoXDTIyMDgy
NTAxMjYyOVowNjEnMCUGA1UEAwwedmNlbnRlci5pbnRlcm5hbC5jYXJsb3NlZHAu
Y29tMQswCQYDVQQGEwJVUzCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEB
ANU7somiLvW76EFF/VDJ5EUJ7uD6RDOhXz24Q/VkVnT/w00tif4Z2wcSxqjhirBe
3iY4eh+LqKawIGXlwJjmJXrUGgZ2PoC3bw+Jx9NIDJfhB776R8Hz5KTeEbsECMVD
GtyYrqErUMhGNh9eIaYqv3us3LdPVltkeqQhvVBnzMJyb9wtXRYebNfD86dZVWWU
FMOcf2C1jnlGS5HYPgl9sirwDQkfBbRI76BVoUGScKrLDqRZ7kfdsHf4nKHqzni3
1SuyWXX8ehMYpokpIAnsRTrqg5AhMQGTKDbJ/jUYvsK3IfyCheenUE+87dYR5AfC
0ilK/LRQGuTx/FRsc4pHRT0CAwEAAaOByzCByDALBgNVHQ8EBAMCA6gwKQYDVR0R
BCIwIIIedmNlbnRlci5pbnRlcm5hbC5jYXJsb3NlZHAuY29tMB0GA1UdDgQWBBQQ
5wd2LdOqn3bCYWylVRn82fiZcjAfBgNVHSMEGDAWgBTEwwPyUydRdTYpfWrfYEdk
7QawXTBOBggrBgEFBQcBAQRCMEAwPgYIKwYBBQUHMAKGMmh0dHBzOi8vdmNlbnRl
ci5pbnRlcm5hbC5jYXJsb3NlZHAuY29tL2FmZC92ZWNzL2NhMA0GCSqGSIb3DQEB
CwUAA4IBAQC00rANWvJ0/zjfyHfrkENrDc7y8w2F/aHUmQep6PYaqjLurEtiQipn
9E/IBnReBumlveWL2LHMzr61Tpr6BB/FmXU1yql0hCEBxSKXO8ZyoZT0GLGZoEXx
xz4b/sgO2carH1dSYsSY5tVZBM1x+qI09ptkY5EsRugfwN7nWa4NDCOV0olMat6Q
5LTAXR2r2l0n79T5R6NCYPpdIBWxSXYVU79gJ+S0ttBED10YqsuOLYL44ptWnNHN
KTTJ/7qcIrN1tA61Xu5vQ3qLvGC99lLKfHieAOF7fIQaRBQuKKvk7fh1iP08UZR7
iO7C7xg5Xt2fUpq7ksVEpzb+NzF5H6ux
-----END CERTIFICATE-----
---
Server certificate
subject=/CN=vcenter.internal.carlosedp.com/C=US
issuer=/CN=CA/DC=internal/DC=carlosedp.com/C=US/ST=California/O=vcenter.internal.carlosedp.com/OU=VMware Engineering
---
No client certificate CA names sent
Server Temp Key: ECDH, P-256, 256 bits
---
SSL handshake has read 1734 bytes and written 322 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES256-GCM-SHA384
    Session-ID: 64E45FCA2B9FD4F4CF7398B9D91AE63C95CDA5B9F27AF912ECF983C6FDC036B5
    Session-ID-ctx:
    Master-Key: 57D6125386BEDE13E4ACBB667B3DC4810C7968AF4CDB283D069974EA0899A66290D56464CB02C77AC71ED438F7DB53DA
    TLS session ticket lifetime hint: 300 (seconds)
    TLS session ticket:
    0000 - 8f 8d 01 81 ca 84 3c 2e-38 63 e2 1b 84 ef 36 4e   ......<.8c....6N
    0010 - b6 f4 c2 3a 7b e7 f0 58-5a 27 28 57 33 b5 d2 37   ...:{..XZ'(W3..7
    0020 - c3 ea ab 06 77 b0 f6 db-0d f6 01 86 f7 4b 9b 32   ....w........K.2
    0030 - 2c 42 0b aa dd d8 7b 46-d1 0d 06 81 c7 b1 f5 23   ,B....{F.......#
    0040 - 01 8a 0e d1 8f 00 85 da-a7 58 49 c5 1c 56 37 8d   .........XI..V7.
    0050 - 83 c4 c6 1d 89 64 7c 2e-c2 31 55 74 1a d0 33 07   .....d|..1Ut..3.
    0060 - 50 d8 d7 ab 8c f5 4d ed-52 92 7a ea 16 d2 8e 89   P.....M.R.z.....
    0070 - 25 15 2a d0 95 da 54 d5-b2 96 a0 99 fc 14 6e 61   %.*...T.......na
    0080 - 10 e6 c1 fa ea e2 7f bb-a4 36 9a a5 3a cf 6e fb   .........6..:.n.
    0090 - 77 c9 d5 de 43 cd 76 46-d9 d1 bf 6c c0 d2 f1 67   w...C.vF...l...g

    Start Time: 1659620186
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
---

```

**Comment by jcpowermac** [comment]
@carlosedp I am pretty sure this is the cause: https://github.com/golang/go/issues/51991

**Comment by carlosedp** [comment]
I didn't get the fix from reading the issue... it means the problem is in Go, in MacOS or in the installer code? :)

**Comment by BlaineEXE** [comment]
@carlosedp did you resolve this? And how? I'm hitting it also

**Comment by carlosedp** [comment]
I haven't tried again. My lab cluster is still up and I haven't tried to install it again.

**Comment by ka1ne** [comment]
@BlaineEXE did you get past this?

**Comment by jcpowermac** [comment]
The problem is the ESXi host certificate validity time is too long. If you change the vCenter configuration to change the date less then a year and then redeploy all the certificates the installer will have no issues.

**Comment by BlaineEXE** [comment]
> @BlaineEXE did you get past this?

I got past it by having a Jenkins server do the deployment for me. I believe that JC's answer is the only real way to solve that, but since I'm not an admin of the vCenter instance and ours was on leave, I couldn't verify that.

**Comment by kulkarma** [comment]
Hi,

I am trying to deploy openshift 4.10.37 and 4.11.9 installer.  I see the following error:
govc import.ova -options=./rhcos.json -name=rhcos-4119  -pool=/Infra-DnD/Datacenter/host/GTC-Cluster/Resources ~/ocp410/rhcos-4.11.9-x86_64-vmware.x86_64.ova
[23-07-23 07:26:38] Uploading disk.vmdk... Error: Post "https://10.254.70.6/nfc/52811492-70b8-9008-8a33-5395f95f67ef/disk-0.vmdk": read tcp 10.254.71.6:56322->10.254.70.6:443: read: connection reset by peer
govc: Post "https://10.254.70.6/nfc/52811492-70b8-9008-8a33-5395f95f67ef/disk-0.vmdk": read tcp 10.254.71.6:56322->10.254.70.6:443: read: connection reset by peer

I have logged into vcenter as administrator of vcenter still shows this problem. The same was working earlier I verified connectivity and verified the certitifcae validity which is intact with OK status.
 

regards
Madhu
